### ISSUE TITLE
NLHE crashed

### ISSUE THREAD

**Issue by ctm** [ctm/mb2-doc#844]
Find and fix whatever killed event #3031.

```
2022-02-08T01:49:32.177894617Z thread '<unnamed>' panicked at 'index out of bounds: the len is 0 but the index is 0', lib/commands/src/game.rs:243:46
```
This was due to the way all-ins on vacation are handled:
```
Dealing #242586: 1000 2000 No-Limit Texas Hold’em (High only)
🍾grizz 🐻 antes 975 as the big blind and is all-in
🎴NNR13 blinds 1000
🍾grizz 🐻 is all in and does not post a blind
🍾grizz 🐻 folds due to being on vacation
1    [🍾grizz 🐻]        0 975 [   0] 
3   > gerdog           48250         0  [  ] [  ]
4     deadhead         51100         0  [  ] [  ]
6     fryday           46075         0  [  ] [  ]
7     ADBDaVoice 🎤     55150         0  [  ] [  ]
8 B   ODB Phat Mack🚀🚀  41950         0  [  ] [  ]
9     🎴NNR13           25500      1000  [  ] [  ]
My cards are: [Tc] [5h]
gerdog folds
It is 2000 to you
deadhead folds
fryday folds
ADBDaVoice 🎤 folds
ODB Phat Mack🚀🚀 raises 5975 to 7975
🎴NNR13 folds
Uncalled 6975 returned to ODB Phat Mack🚀🚀
```
The hand was over before certain things were set up and when it was time to show who won and why, it died trying to access the stuff that wasn't set up.

I hate creating bugs, but I'm super lucky to have such an understanding user base.

**Comment by ctm** [comment]
Although it's a shame to crash a tourney that people were playing in, what killed it is sort of funny.

Back on August 14th, 2020 I created a new "pot splitter" that was both simpler than the original pot splitter and also handled "the sevens rule" for CA lowball. Even though the new version was simpler, I was afraid that the new version might have bugs, so I wrote code that would split the pot both with the old splitter and the new splitter and if there was a difference, report that difference, but use the original pot splitter's choice, because&dash;at the time&mdash;it had been running for long enough that it was considered trustworthy.

Well, when I changed how vacation works, I changed the new pot splitter to understand that a solo ante (i.e. a button or big blind ante) could be given up by a player in vacation, but I didn't change the old pot splitter, because by that time I had forgotten that I run both but keep the results of the original.

So, since the new splitter has been working fine for a year and a half, I'm going to remove the old pot splitter and simply use the new. That will fix this bug.
